otherlibs/unix/unix.ml

```diff
@@ -203,7 +203,68 @@ type wait_flag =
 external execv : string -> string array -> 'a = "unix_execv"
 external execve : string -> string array -> string array -> 'a = "unix_execve"
 external execvp : string -> string array -> 'a = "unix_execvp"
-external execvpe : string -> string array -> string array -> 'a = "unix_execvpe"
+external execvpe_c :
+            string -> string array -> string array -> 'a = "unix_execvpe"
+
+let execvpe_ml name args env =
+  (* Try to execute the given file *)
+  let exec file =
+    try
+      execve file args env
+    with Unix_error(ENOEXEC, _, _) ->
+      (* Assume this is a script and try to execute through the shell *)
+      let argc = Array.length args in
+      (* Drop the original args.(0) if it is there *)
+      let new_args = Array.append
+        [| "/bin/sh"; file |]
```

Comment:
shouldn't we at least try to call `confstr(_CS_PATH)` or it will complicate things too much?

Reply:
There is no OCaml binding for `confstr` so far, and quite frankly I don't miss it.  The OpenBSD implementation of `execvpe` doesn't care either, using
```
#define	_PATH_BSHELL	"/bin/sh"
```
as the shell to execute.